### PR TITLE
Add salt to wallet id hash for ledger accounts

### PR DESCRIPTION
**Pull request by danimoh**
Proposal for salting the public key before the wallet id computation.
Not entirely sure whether we get a lot of upside from doing so, but at the same time it also doesn't hurt.

### CODE REVIEW

File: src/lib/LedgerApi.ts [hunk 1]
```diff
@@ -385,6 +385,7 @@ class LedgerApi {
     }
 
     // private fields and methods
+    private static readonly WALLET_ID_SALT = 42;
```
Comment (sisou) [inline src/lib/LedgerApi.ts]: A fixed salt still allows a pre-computed rainbow attack.

Comment (danimoh) [reply src/lib/LedgerApi.ts]: Rainbow tables are not really feasible here. The hash function is still fed 32 bytes of random data.
But no one would use a rainbow table here anyways. What's more of a problem is, that the wallet ID is easy to calculate if the public key is known.
What are you suggesting? To create a random salt and store that in a database in the hub along with the hashes? Do you _really_ think that's worth the effort?

Comment (sisou) [reply src/lib/LedgerApi.ts]: No, I think hashing twice is better.

Comment (nibhar) [reply src/lib/LedgerApi.ts]: Double hashing does not prevent the walletId being computable once the public key is known. A random salt stored in the database does. So if we decide that it should not be computable, then a random salt should be used in my opinion.

**Comment by danimoh** [comment]
Closing this in favor of https://github.com/nimiq/hub/pull/298

**Pull request closed without merging**